eventdisplay/tracks/FairEveTracks.cxx

```diff
@@ -0,0 +1,99 @@
+/********************************************************************************
+ *    Copyright (C) 2020 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+/*
+ * FairTracks.cxx
+ *
+ *  Created on: 4 maj 2020
+ *      Author: Daniel Wielanek
+ *		E-mail: daniel.wielanek@gmail.com
+ *		Warsaw University of Technology, Faculty of Physics
+ */
+#include "FairEveTracks.h"
+
+#include "FairEveRecoTrackList.h"
+#include "FairEveTrack.h"
+
+#include <TEveElement.h>
+#include <TEveManager.h>
+#include <TEveSelection.h>
+#include <TEveTrack.h>
+
+FairEveTracks::FairEveTracks(Bool_t acceptCompound)
+    : fEventManager(nullptr)
+    , fEveTrList(nullptr)
+    , fPt{0, 10}
+    , fEta{-10, 10}
+    , fEnergy{0, 10}
+    , fUsePt(kFALSE)
+    , fUseEta(kFALSE)
+    , fUseEnergy(kFALSE)
+    , fAcceptCompound(acceptCompound)
+{
+    fEveTrList = new TObjArray();
+}
+
+void FairEveTracks::ResetGroup()
+{
+    TEveElement *el = dynamic_cast<TEveElement *>(this);
+    if (el == nullptr)
+        return;
+    for (Int_t i = 0; i < fEveTrList->GetEntriesFast(); i++) {
+        TEveTrackList *ele = static_cast<TEveTrackList *>(fEveTrList->At(i));
+        gEve->RemoveElement(ele, el);
+    }
+    fEveTrList->Clear();
+}
+
+void FairEveTracks::SwapTracks()
+{
+    for (int i = 0; i < fEveTrList->GetEntriesFast(); i++) {
+        TEveTrackList *eveList = (TEveTrackList *)fEveTrList->UncheckedAt(i);
+        for (TEveElement::List_i iter = eveList->BeginChildren(); iter != eveList->EndChildren(); ++iter) {
+            FairEveTrack *el = dynamic_cast<FairEveTrack *>((*iter));
+            if (el->GetRnrSelf()) {
+                el->SetRnrSelfChildren(kFALSE, kFALSE);
+            } else {
+                el->SetRnrSelfChildren(kTRUE, kTRUE);
+            }
+        }
+    }
+    gEve->Redraw3D(kFALSE);
+}
+
+InitStatus FairEveTracks::Init()
+{
+    fEventManager = FairEventManager::Instance();
+    return kSUCCESS;
+}
+
+FairEveTracks::~FairEveTracks() {}
+
+TEveTrackList *FairEveTracks::GetTrackGroup(TString groupName, Color_t color)
```

Comment:
You are right, this cannot be a const member function. Let's maybe rename it then to `AddTrackGroup` or `CreateTrackGroup` to not confuse it with the Getter pattern. What do you think?

Reply:
Btw, what happens, if we call `GetTrackGroup(TString groupName, Color_t color)` twice with the same arguments? Do you want it to create the same track group twice or do you want it to return the first?

Reply:
When I call it second time then it should return the old group. All "similar" particles should belong to the same group (e.g. pions)
![Bez nazwy](https://user-images.githubusercontent.com/8748256/92023181-afc63c80-ed5c-11ea-9713-897a874eff43.png)
).

Reply:
For no I called this "FindTrackGroup".

Reply:
I have more questions:
1. Does a `FairEveTracks` object only ever store "track groups" (`TEveTrackList` (derived) objects) on its first level of children?
   a. If yes, have you considered implementing `FindTrackGroup()` with [`TEveElement::FindChild()`](https://root.cern/doc/master/classTEveElement.html#aee28f791af457d9397cb739b3355fbb1)?
   b. If no, and we really have to distinguish "track groups" from other kinds of children, then the follow-up question would be: Why do you not use a map container for caching the access to the "track groups" by name?
2. The `SwapTracks()` naming confused me, usually it means that objects are replaced which is not the case here.
   a. I suggest renaming it to something like `ToggleTracks()`, `ToggleTracksVisibility()` or even `ToggleTrackGroups()` if it is meant to operate on the "track group" level only.
   b. Your current implementation of `SwapTracks()` seems to toggle visibility on each individual track within a "track group". Is this intended? If not, would setting `SetRnrSelfChildren()` on the "track group" level work, too?

Reply:
1. Fixed
2. Replaced by ToogleTracks
This is designed to work at track level - toggling group can be done by clicking, in case of tracks - it's not so easy because you have few hundreds of them.
However toggle groups is not a bad idea - I added it to FairGeoTracks. I also removed virtual label from FairEveTracks::ToggleTracks - if we operate only on tracks there is no point to make this method virtual.